packages/headless/src/controllers/commerce/context/cart/headless-cart.ts

```diff
@@ -171,8 +214,18 @@ export function buildCart(engine: CommerceEngine, props: CartProps = {}): Cart {
     },
 
     updateItem(item: CartItem) {
-      if (isNewQuantityDifferent(item.productId, item.quantity)) {
-        // TODO LENS-1497: log ec.cartAction; if new quantity > previous, 'add', otherwise, 'remove'.
+      const prevItem = itemSelector(getState(), item.productId);
+      const doesNotNeedUpdate = !prevItem && item.quantity <= 0;
+
+      if (isEqual(item, prevItem) || doesNotNeedUpdate) {
```

Comment:
`doesNotNeedUpdate` is already computed here as a bool, so we should first assert its value before checking the more 'lengthy' compare

```suggestion
      if (doesNotNeedUpdate || isEqual(item, prevItem)) {
```